### PR TITLE
feat: generate all accents with whiskers

### PR DESCRIPTION
**Pull request by arminius-smh**
Hello,

I wrote a little script that adds accent colors to the flavors, which would also close https://github.com/catppuccin/fcitx5/issues/3

The script only changes the `Highlightcolor` and `HighlightCandidateColor` to all the color accents from the flavors (together with the comment above each of those, describing the color name) and appends the accent name to the `Name` field

With this, catppuccin-*flavor*-peach and catppuccin-*flavor* are identically, but I still left them for compatibility reasons

Here my scripts to see a little bit better if I have messed something up:
<details> <summary>accents.sh</summary>

```sh
#!/usr/bin/env bash

declare -A color_map_latte
declare -A color_map_frappe
declare -A color_map_macchiato
declare -A color_map_mocha

color_map_latte=(
    [rosewater]="#dc8a78"
    [flamingo]="#dd7878"
    [pink]="#ea76cb"
    [mauve]="#8839ef"
    [red]="#d20f39"
    [maroon]="#e64553"
    [peach]="#fe640b"
    [yellow]="#df8e1d"
    [green]="#40a02b"
    [teal]="#179299"
    [sky]="#04a5e5"
    [sapphire]="#209fb5"
    [blue]="#1e66f5"
    [lavender]="#7287fd"
)

color_map_frappe=(
    [rosewater]="#F2D5CF"
    [flamingo]="#EEBEBE"
    [pink]="#F4B8E4"
    [mauve]="#CA9EE6"
    [red]="#E78284"
    [maroon]="#EA999C"
    [peach]="#EF9F76"
    [yellow]="#E5C890"
    [green]="#A6D189"
    [teal]="#81C8BE"
    [sky]="#99D1DB"
    [sapphire]="#85C1DC"
    [blue]="#8CAAEE"
    [lavender]="#BABBF1"
)

color_map_macchiato=(
    [rosewater]="#F4DBD6"
    [flamingo]="#F0C6C6"
    [pink]="#F5BDE6"
    [mauve]="#C6A0F6"
    [red]="#ED8796"
    [maroon]="#EE99A0"
    [peach]="#F5A97F"
    [yellow]="#EED49F"
    [green]="#A6DA95"
    [teal]="#8BD5CA"
    [sky]="#91D7E3"
    [sapphire]="#7DC4E4"
    [blue]="#8AADF4"
    [lavender]="#B7BDF8"
)

color_map_mocha=(
    ["rosewater"]="#F5E0DC"
    ["flamingo"]="#F2CDCD"
    ["pink"]="#F5C2E7"
    ["mauve"]="#CBA6F7"
    ["red"]="#F38BA8"
    ["maroon"]="#EBA0AC"
    ["peach"]="#FAB387"
    ["yellow"]="#F9E2AF"
    ["green"]="#A6E3A1"
    ["teal"]="#94E2D5"
    ["sky"]="#89DCEB"
    ["sapphire"]="#74C7EC"
    ["blue"]="#89B4FA"
    ["lavender"]="#B4BEFE"
)

create_accents() {
    local color_flavor="$1"
    local -n color_map=$2

    for key in "${!color_map[@]}"; do
        cp -r "./fcitx5/src/catppuccin-$color_flavor/" "./fcitx5/src/catppuccin-$color_flavor-$key"
        local new_HighlightCandidateColor=${color_map[$key]}
        local new_HighlightColor=${color_map[$key]}

        sed -i \
            -e "3s/$/ ${key^}/" \
            -e "16s/.*/#${key^}/" \
            -e "19s/.*/#${key^}/" \
            -e "s/^\(HighlightCandidateColor=\).*/\1$new_HighlightCandidateColor/" \
            -e "s/^\(HighlightColor=\).*/\1$new_HighlightColor/" \
            "./fcitx5/src/catppuccin-$color_flavor-$key/theme.conf"
    done
}

create_accents "latte" color_map_latte
create_accents "frappe" color_map_frappe
create_accents "macchiato" color_map_macchiato
create_accents "mocha" color_map_mocha
```

### CODE REVIEW

**Comment by sgoudham** [comment]
Hi! Thanks for the PR but unfortunately this method of generating accents is outdated.

We'd be looking to update this repository into using [catppuccin/whiskers](https://github.com/catppuccin/whiskers) which will allow you to trivially generate these accents and files.

Would you be interested in taking on this work?

File: justfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+_default:
+  @just --list
+
+flavors := "latte frappe macchiato mocha"
+accents := "rosewater flamingo pink mauve red maroon peach yellow green teal sky sapphire blue lavender"
```
Comment (uncenter) [inline justfile:5]: Instead of these, use `whiskers --list-flavors` and `whiskers --list-accents`.

File: justfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+build:
+  @for flavor in {{flavors}}; do \
+    whiskers -f "$flavor" "./fcitx5.tera"; \
+    for accent in {{accents}}; do \
+      cp "./assets/$flavor-radio.png" "./src/catppuccin-$flavor-$accent/radio.png"; \
+      cp "./assets/$flavor-arrow.png" "./src/catppuccin-$flavor-$accent/arrow.png"; \
+    done; \
+  done
```
Comment (uncenter) [inline justfile:14]: You are using a matrix for the Whiskers build process, so you don't need to run Whiskers for each flavor in this loop.
```suggestion
  #!/usr/bin/env bash
  whiskers fcitx5.tera
  whiskers --list-flavors -o plain | while read -r flavor; do
    whiskers --list-accents -o plain | while read -r accent; do
      cp "./assets/$flavor-radio.png" "./src/catppuccin-$flavor-$accent/radio.png"
      cp "./assets/$flavor-arrow.png" "./src/catppuccin-$flavor-$accent/arrow.png"
    done
  done
```

File: justfile (hunk 1 above)
Comment (uncenter) [reply justfile:5]: I've edited the script below, haven't tested it but it should work.

File: justfile (hunk 2 above)
Comment (arminius-smh) [reply justfile]: Thanks, this looks way cleaner

File: justfile [hunk 3]
```diff
@@ -0,0 +1,12 @@
+_default:
+  @just --list
+
+build:
+  #!/usr/bin/env bash
+  whiskers fcitx5.tera
+  whiskers --list-flavors -o plain | while read -r flavor; do
+    whiskers --list-accents -o plain | while read -r accent; do
```
Comment (uncenter) [inline justfile:8]:
```suggestion
  for flavor in $(whiskers --list-flavors -o plain ); do
    for accent in $(whiskers --list-accents -o plain ); do
```
Oops, I think this simpler for loop should work!

Comment (arminius-smh) [reply justfile]: Right, then the simpler loop it shall be!

**Comment by arminius-smh** [comment]
I also noticed and fixed typo in the preview, hello in japanese is こんにち**は** and not こんにち**わ**

**Review by sgoudham — CHANGES_REQUESTED** [review]
Hey 👋,
Thanks for updating this PR to use whiskers.

I see that Fcitx5 is only available on Unix systems so I feel like there's a stronger case to symlink the png files into each flavour from the assets directory instead of duplicating the files across the entire repository? 

I'd be happy to merge if the PR was updated to use symlinks, but also open to discussion if you think that's not a good idea!

**Comment by arminius-smh** [comment]
> I see that Fcitx5 is only available on Unix systems so I feel like there's a stronger case to symlink the png files into each flavour from the assets directory instead of duplicating the files across the entire repository?

I wouldn't mind - but I don't think I quite understand?

Do you want them to be symlinked like this?
```bash
ln -s "../../assets/$flavor-radio.png" "./src/catppuccin-$flavor-$accent/radio.png"
ln -s "../../assets/$flavor-arrow.png" "./src/catppuccin-$flavor-$accent/arrow.png"
```

The themes have to be moved to $XDG_DATA_HOME/fcitx5/themes/ though and can't be selected from the menu, and thus this doesn't make much sense or am I missing something?

**Comment by sgoudham** [comment]
> The themes have to be moved to $XDG_DATA_HOME/fcitx5/themes/ though and can't be selected from the menu, and thus this doesn't make much sense or am I missing something?

Ah yeah, I forgot that the symlinks would be broken once the theme is copied over. I was just a little concerned that we're duplicating the same asset 56 times across the repository. The PNGs aren't huge so I'm fine with approving but figured I'd mention it anyways.

**Comment by sgoudham** [comment]
Actually, I haven't tried this before so I'm unsure if the symlink would break.

If a person clones the repository and then runs a `cp` command to transfer a directory to somewhere else in their local filesystem, I would have thought the symlinked file is still copied? Or is that just not the case.

**Comment by arminius-smh** [comment]
Yeah if the original pngs are copied over as well in the same relative structure then these wouldn't break

So, if you really want symlinks there are two options
1) have the users copy the assets directory to `$XDG_DATA_HOME/fcitx5/assets` so that the links in `$XDG_DATA_HOME/fcitx5/themes/catppuccin-flavor-accent` would resolve to `../../assets/flavor-radio.png`
- although I wouldn't really like that option, as this seems a bit of an weird thing to do, having to copy the assets directory in an odd place

2) move the relevant pngs to src/assets/ so that the symlinks would resolve too `../assets/flavor-radio.png`, so that there still is only `cp -r ./fcitx5/src/* ~/.local/share/fcitx5/themes` to do as in the readme right now, then there would just have to be an assets directory in the fcitx5 themes folder `$XDG_DATA_HOME/fcitx5/themes/assets`

- I guess technically I could also instruct users to update the symlink path themselves after cloning the git repo, but I think that extra step would be a bit cubersome 

Only, in both cases users couldn't just do `cp -r ./fcitx5/src/catppuccin-mocha-mauve ~/.local/share/fcitx5/themes`, this would break the relative symlinks, the assets directory would have to be copied over as well to the correct place

If there is any of those options you prefer, I can do that

**Comment by sgoudham** [comment]
I see, thanks for taking the time to explain it!

I think it's fine in this case since the files are so small, happy to merge and thanks again for your work!

**Review by sgoudham — APPROVED** [review]

**Pull request merged**